credentials/src/main/kotlin/web5/sdk/credentials/PresentationExchange.kt

```diff
@@ -20,40 +28,77 @@ public object PresentationExchange {
   }
 
   /**
-   * Checks if the given [presentationDefinition] is satisfied based on the provided input descriptors and constraints.
+   * Validates whether a verifiable credential (VC) satisfies a given Presentation Definition.
    *
-   * @param presentationDefinition The Presentation Definition to be evaluated.
-   * @return `true` if the Presentation Definition is satisfied, `false` otherwise.
-   * @throws UnsupportedOperationException if certain features like Submission Requirements or Field Filters are not implemented.
+   * @param vcJwt The VC in JWT format.
+   * @param presentationDefinition The Presentation Definition to be satisfied.
    */
   public fun satisfiesPresentationDefinition(
-    credential: VerifiableCredential,
+    vcJwt: String,
```

Comment:
I its because ssome of the filter could be for the actual payload of the vc
$.iss is the correct way to filter by issuer not vc.issuer (this will be removed as per spec)

Reply:
Context starting here: https://sq-tbd.slack.com/archives/C05SXUR39FE/p1698361441821219?thread_ts=1698337236.592429&cid=C05SXUR39FE

Reply:
agree w/ the concern, but it's better to decouple the logic IMO

Reply:
Sounds like you're advocating for leaving as is. I remain cautious about the approach. Agree with the intent of decoupling verification logic from validation.

My proposal is to stick to the following convention:
* When a function is dealing with data that's assumed to be trusted, use the `VerifiableCredential` type.
* When a function is dealing with data that needs to be verified, use the secured representation (i.e. `vcJwt` or DI).

For this function, the data is assumed to be trusted already. Given that, it would fall to the first bucket. 
On the other hand, the `VerifiableCredential.verify` function would follow the second pattern.


Alternatively, we could define types to represent:
* A secured verifiable credential: `JwVerifiableCredential` and `DataIntegrityVerifiableCredential`.
* An unsecured verifiable credential `UnsecuredVerifiableCredential`.

This would make it very difficult for users to screw up. 

Either of these two would be more beneficial IMHO. Those are my suggestions, I'll leave it up to @phoebe-lew to decide whether they're useful or not.